tools/loadtesting/terraform/ecs.tf

```diff
@@ -0,0 +1,385 @@
+
+  default_action {
+    target_group_arn = aws_alb_target_group.main.arn
+    type             = "forward"
+  }
+}
+
+resource "aws_alb_listener" "http" {
+  load_balancer_arn = aws_alb.main.arn
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type = "redirect"
+
+    redirect {
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+resource "aws_ecs_cluster" "fleet" {
+  name = "${local.prefix}-backend"
+
+  setting {
+    name  = "containerInsights"
+    value = "enabled"
+  }
+}
+
+resource "aws_ecs_service" "fleet" {
+  name                               = "fleet"
+  launch_type                        = "FARGATE"
+  cluster                            = aws_ecs_cluster.fleet.id
+  task_definition                    = aws_ecs_task_definition.backend.arn
+  desired_count                      = 10
+  deployment_minimum_healthy_percent = 100
+  deployment_maximum_percent         = 200
+  health_check_grace_period_seconds  = 30
+
+  load_balancer {
+    target_group_arn = aws_alb_target_group.main.arn
+    container_name   = "fleet"
+    container_port   = 8080
+  }
+
+  network_configuration {
+    subnets         = module.vpc.private_subnets
+    security_groups = [aws_security_group.backend.id]
+  }
+
+  depends_on = [aws_alb_listener.http, aws_alb_listener.https-fleetdm]
+}
+
+resource "aws_cloudwatch_log_group" "backend" {
+  name              = "fleetdm"
+  retention_in_days = 1
+}
+
+data "aws_region" "current" {}
+
+data "aws_secretsmanager_secret" "license" {
+  name = "/fleet/license"
+}
+
+resource "aws_ecs_task_definition" "backend" {
+  family                   = "fleet"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  execution_role_arn       = aws_iam_role.main.arn
+  task_role_arn            = aws_iam_role.main.arn
+  cpu                      = 1024
+  memory                   = 2048
+  container_definitions = jsonencode(
+    [
+      {
+        name      = "prometheus-exporter"
+        image     = "917007347864.dkr.ecr.us-east-2.amazonaws.com/prometheus-to-cloudwatch:latest"
+        essential = false
+        logConfiguration = {
+          logDriver = "awslogs"
+          options = {
+            awslogs-group         = aws_cloudwatch_log_group.backend.name
+            awslogs-region        = data.aws_region.current.name
+            awslogs-stream-prefix = "fleet-prometheus-exporter"
+          }
+        }
+        environment = [
+          {
+            name  = "CLOUDWATCH_NAMESPACE"
+            value = "fleet-loadtest"
+          },
+          {
+            name  = "CLOUDWATCH_REGION"
+            value = "us-east-2"
+          },
+          {
+            name  = "PROMETHEUS_SCRAPE_URL"
+            value = "http://localhost:8080/metrics"
+          },
+        ],
+      },
+      {
+        name        = "fleet"
+        image       = docker_registry_image.fleet.name
+        cpu         = 1024
```

Comment:
should probably be var?

Reply:
it is a var, terraform pulls and repushes the image var provided.

Reply:
sorry I should have been more clear, I meant the `cpu` value since we might want to parameterize this for different load testing thresholds

Reply:
True, I moved it to be hardcoded since it simplifies the interface for developers who will be more interested in asking "does my code work in loadtest". Its basically to make the distinction between devs familiar with terraform (who can just go and change a single hardcoded value) vs devs who aren't (who are provided with a simpler interface).